students/byeongyeon/westagram/users/views.py

```diff
@@ -1,3 +1,35 @@
-from django.shortcuts import render
-
 # Create your views here.
+from django.shortcuts       import render
+import json
+
+from django.http            import JsonResponse
+from django.views           import View
+
+from users.models           import User
+from users.validator        import validate_email, validate_password, validate_mobile
+
+class RegisterView(View):      
+    def post(self, request):
+        data = json.loads(request.body)
+        try:
+            if User.objects.filter(email=data['email']).exists():
+                return JsonResponse({"message": "Email is already in use"}, status=400)
+            elif not validate_email(data["email"]):
+                return JsonResponse({"message": "Email format is invalid"}, status=400)
+            elif not validate_password(data["password"]):
+                return JsonResponse({"message": "Password format is invalid"}, status=400)
+            elif not validate_mobile(data["mobile"]):
+                return JsonResponse({"message": "Mobile number format is invalid"}, status=400)
+            else:         
+                User.objects.create(
+                    name     = data['name'],
+                    email    = data['email'],
+                    password = data['password'],
+                    mobile   = data['mobile'],
+                    address  = data['address'],
+                    age      = data['age']
+                )
+        except KeyError:
+            return JsonResponse({"message": "Please provide valid information. Check your email, password and mobile number"}, status=400)
+
+        return JsonResponse({"message":"SUCCESS"}, status=201)
```

Comment:
1. except를 retrun 아래로 내려주세요. except는 더 추가 될 수 있고, 로직 흐름상 최종 결과는 try 안에 있는 것이 좋습니다.
2. return message는 front에서 보는 메시지 입니다. 장문의 메시지 보다는 명확한 메시지로 보내 주어야 front에서도 해당 메시지를 parsing해서 로직을 구현하기 편해집니다.
```suggestion
            return JsonResponse({"message":"SUCCESS"}, status=201)
        except KeyError:
            return JsonResponse({"message": "KEY_ERROR"}, status=400)
```

Reply:
리뷰 반영하여 수정하였습니다.